storage/adapter.go

```diff
@@ -0,0 +1,407 @@
+		GasLimit: types.NewInt(1000000),
+		Method:   actors.SMAMethods.PublishStorageDeals,
+		Params:   params,
+	})
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return smsg.Cid(), nil
+}
+
+func (m *StorageMinerNodeAdapter) WaitForSelfDeals(ctx context.Context, publicStorageDealsMsgCid cid.Cid) ([]uint64, uint8, error) {
+	r, err := m.api.StateWaitMsg(ctx, publicStorageDealsMsgCid)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	if r.Receipt.ExitCode != 0 {
+		return nil, 0, nil
+	}
+
+	var resp actors.PublishStorageDealResponse
+	if err := resp.UnmarshalCBOR(bytes.NewReader(r.Receipt.Return)); err != nil {
+		return nil, 0, err
+	}
+
+	return resp.DealIDs, r.Receipt.ExitCode, nil
+}
+
+func (m *StorageMinerNodeAdapter) SendPreCommitSector(ctx context.Context, sectorID uint64, commR []byte, ticket s2.SealTicket, pieces ...s2.Piece) (cid.Cid, error) {
+	dealIDs := make([]uint64, len(pieces))
+	for idx, p := range pieces {
+		dealIDs[idx] = p.DealID
+	}
+
+	params := &actors.SectorPreCommitInfo{
+		SectorNumber: sectorID,
+		CommR:        commR,
+		SealEpoch:    ticket.BlockHeight,
+		DealIDs:      dealIDs,
+	}
+	enc, aerr := actors.SerializeParams(params)
+	if aerr != nil {
+		return cid.Undef, xerrors.Errorf("could not serialize commit sector parameters: %w", aerr)
+	}
+
+	msg := &types.Message{
+		To:       m.maddr,
+		From:     m.waddr,
+		Method:   actors.MAMethods.PreCommitSector,
+		Params:   enc,
+		Value:    types.NewInt(0), // TODO: need to ensure sufficient collateral
+		GasLimit: types.NewInt(1000000 /* i dont know help */),
+		GasPrice: types.NewInt(1),
+	}
+
+	log.Info("submitting precommit for sector: ", sectorID)
+	smsg, err := m.api.MpoolPushMessage(ctx, msg)
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("pushing message to mpool: %w", err)
+	}
+
+	return smsg.Cid(), nil
+}
+
+func (m *StorageMinerNodeAdapter) SendProveCommitSector(ctx context.Context, sectorID uint64, proof []byte, dealIDs ...uint64) (cid.Cid, error) {
+	// TODO: Consider splitting states and persist proof for faster recovery
+	params := &actors.SectorProveCommitInfo{
+		Proof:    proof,
+		SectorID: sectorID,
+		DealIDs:  dealIDs,
+	}
+
+	enc, aerr := actors.SerializeParams(params)
+	if aerr != nil {
+		return cid.Undef, xerrors.Errorf("could not serialize commit sector parameters: %w", aerr)
+	}
+
+	msg := &types.Message{
+		To:       m.maddr,
+		From:     m.waddr,
+		Method:   actors.MAMethods.ProveCommitSector,
+		Params:   enc,
+		Value:    types.NewInt(0), // TODO: need to ensure sufficient collateral
+		GasLimit: types.NewInt(1000000 /* i dont know help */),
+		GasPrice: types.NewInt(1),
+	}
+
+	// TODO: check seed / ticket are up to date
+
+	smsg, err := m.api.MpoolPushMessage(ctx, msg)
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("pushing message to mpool: %w", err)
+	}
+
+	return smsg.Cid(), nil
+}
+
+func (m *StorageMinerNodeAdapter) WaitForProveCommitSector(ctx context.Context, proveCommitSectorMsgCid cid.Cid) (uint8, error) {
```

Comment:
Ditto about events, we want to be able to notify the state machine about message reverts